site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+
+See [Navigating with keyboard section below](/#navigating-with-keyboard) for more information on helping the user navigate forms with keyboards.
+
+<Image src="../../static/patterns/form/form-states-focus@2x.png" alt="Focus indicator" style="max-width:376px;" viewable /> 
+
+---
+
+### Empty/unselected and filled/selected
+Form controls basic function is to allow the user enter an value either by input or selection.
+
+<Image src="../../static/patterns/form/form-states-filled@2x.png" alt="Empty and filled text fields" style="max-width:376px;" viewable /> 
+
+<Image src="../../static/patterns/form/form-states-selected@2x.png" alt="Unselected and selected form controls" style="max-width:376px;" viewable /> 
+
+
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <p><strong>Help the user by minimising unnecessary and redundant work needed for filling the form. </strong>Especially long forms can become tedious to fill, especially, if the same information must be entered multiple times.</p>
+    <ul style="margin-bottom:0;">
+        <li><strong>Provide a pre-selected default option</strong> if some option is known to be the most commonly used.</li>
+        <li><strong>Pre-populate known values if data is available.</strong>
+            <ul>
+                <li>The user <em>should</em> not be made to input the same information twice.</li>
+            </ul>
+        </li>
+        <li><strong>Enable autocompleting values</strong> to provide automated assistance in filling out form field values.
+            <ul>
+                <li>An appropriate autocomplete attribute <em>should</em> be deigned for fields if possible.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+<div class="guideline-dont" style="background: var(--color-error-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-dont-label" style="color:var(--color-error); margin-bottom:var(--spacing-s);"><IconError style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Don't</strong></div>
+    <p><strong>Selections that demand the users consent, like approving terms and conditions <em>must</em> never be pre-selected as default.</strong></p>
+</div>
+
+---
+
+## Providing instructions
+To help the user enter the **right information in the right format**, guidance can be provided on different levels:
+
+1. **Form main and sub-heading** summarise the purpose of the form
+2. **Form wide instructions** to help the user to better understand the functionality of the form
+3. **Contextual instructions** for helping the user to fill specific form controls
+
+---
```

Comment:
> viiva

Tässä tällä `<hr>`:llä on taas vähän päinvastainen vaikutus. Allaoleva sisältö kuitenkin liittyy tähän, mutta jotenkin tämä viiva erottaa näitä turhaan toisistaan.